### PR TITLE
One resource entry per resource

### PR DESCRIPTION
**Pull request by yoavweiss**
This is a resubmission of @plehegar's https://github.com/w3c/web-platform-tests/pull/2919
It tests that a single entry is added per resource, even if requested multiple times.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/254367910)

# Firefox (nightly) #
Testing web-platform-tests at revision 80016c66db9f5bfeb102222103916db04089b346
Using browser at version BuildID 20170716100258; SourceStamp b4e656e5a996dd385e9af43a4d9e207553377c51
Starting 10 test iterations
## Unstable results ##


|                        Test                       |                Subtest                 |            Results            |     Messages     |
|---------------------------------------------------|----------------------------------------|-------------------------------|------------------|
| `/resource-timing/single-entry-per-resource.html` |                                        | **OK: 5/10, TIMEOUT: 5/10**   |                  |
| `/resource-timing/single-entry-per-resource.html` | `Only one resource entry per resource` | **PASS: 5/10, TIMEOUT: 5/10** | `Test timed out` |

## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6567/resource-timing/single-entry-per-resource.html">/resource-timing/single-entry-per-resource.html</a></summary>


|                Subtest                 |            Results            |     Messages     |
|----------------------------------------|-------------------------------|------------------|
|                                        | **OK: 5/10, TIMEOUT: 5/10**   |                  |
| `Only one resource entry per resource` | **PASS: 5/10, TIMEOUT: 5/10** | `Test timed out` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/254367911)

# Sauce (safari) #
Testing web-platform-tests at revision 80016c66db9f5bfeb102222103916db04089b346
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6567/resource-timing/single-entry-per-resource.html">/resource-timing/single-entry-per-resource.html</a></summary>


|                Subtest                 | Results |                  Messages                  |
|----------------------------------------|---------|--------------------------------------------|
|                                        | OK      |                                            |
| `Only one resource entry per resource` | FAIL    | `Can't find variable: PerformanceObserver` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/254367917)

# Chrome (unstable) #
Testing web-platform-tests at revision 32aebcfb57e816c755cbcdd9b3a708ad52ad82f2

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/254367918)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 32aebcfb57e816c755cbcdd9b3a708ad52ad82f2

File: resource-timing/single-entry-per-resource.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+<script>
+  async_test(function (t) {
+    function path(pathname) {
+      var filename = pathname.substring(pathname.lastIndexOf('/')+1);
+      return pathname.substring(0, pathname.length - filename.length);
+    }
+    var img_location = document.location.origin + path(document.location.pathname)
+       + "resources/blue.png";
+    var observer = new PerformanceObserver(
+        t.step_func(function (entryList, obs) {
```
Comment (igrigorik) [inline resource-timing/single-entry-per-resource.html]: Perhaps unlikely, but this test wouldn't catch an implementation that fires two PO callbacks, one for each img, right? Would it make sense to accumulate the entries and verify at onload that there is only one?

**Comment by sideshowbarker** [comment]
w3c-test:mirror

File: resource-timing/single-entry-per-resource.html (hunk 1 above)
Comment (igrigorik) [inline resource-timing/single-entry-per-resource.html]: *bump* 

@yoavweiss wdyt?

Comment (yoavweiss) [inline resource-timing/single-entry-per-resource.html]: Good call. Changed the time PO is registered, the time images are discovered and the time we check to see only a single PO triggered...

**Comment by yoavweiss** [comment]
@igrigorik - PTAL

**Review by igrigorik — APPROVED** [review]
looks great, thanks Yoav! 👍

**Pull request merged**